### PR TITLE
Fix-Authentication-Parsers

### PR DESCRIPTION
**Pull request by KustoKing**
Required items, please complete
   
   Change(s):
   - Added EquivalentBuiltInParser parameter in Template, EventUid column and fixed formatting where needed

   Reason for Change(s):
   - Missing EquivalentBuiltInParser  parameter, EventUid column and cleaned formatting

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by v-atulyadav** [comment]
Hi @KustoKing,
Thanks for your contribution. 
We will provide you an update shortly, meanwhile can you please resolve branch conflicts.

**Comment by v-atulyadav** [comment]
Hi @KustoKing,
Can you please resolve branch conflicts. We will review the Pull Request internally and get back to you shortly.
Thanks

**Comment by v-atulyadav** [comment]
Hi @KustoKing, 
There are still branch conflicts and kql validation is failing. Below are some suggestions for resolving the KQL validation issue.
1. A comma (`,`) is missing in line 33 yellow marked.
2. The comma should be removed from line no 42 red marked
image

**Comment by KustoKing** [comment]
> Referenc

Thanks for your feedback. It should be sorted now

**Comment by v-atulyadav** [comment]
Hi @KustoKing,
Thank you for accommodating the suggested changes! You will receive an update shortly.
Thanks

**Comment by vakohl** [comment]
ETA for this review is 14th Aug

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationAADNonInteractive.yaml [hunk 1]
```diff
@@ -15,23 +15,27 @@ References:
 Description: |
     This ASIM parser supports normalizing Azure Active Directory Non Interactive sign in logs, stored in the AADNonInteractiveUserSignInLogs table, to the ASIM Authentication schema.
 ParserName: ASimAuthenticationAADNonInteractiveUserSignInLogs
+EquivalentBuiltInParser: _ASim_Authentication_AADNonInteractiveUserSignInLogs
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationAADNonInteractive.yaml:18]: @KustoKing Are these changes due to positions changing, is there anything added new or got removed?
Other than EquivalentBuiltInParser property?

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationAADServicePrincipalSignInLogs.yaml [hunk 2]
```diff
@@ -15,6 +15,7 @@ References:
 Description: |
   This ASIM parser supports normalizing Azure Active Directory Service Principal sign in logs, stored in the AADServicePrincipalSignInLogs table, to the ASIM Authentication schema.
 ParserName: ASimAuthenticationAADServicePrincipalSignInLogs
+EquivalentBuiltInParser: _ASim_Authentication_AADServicePrincipalSignInLogs
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationAADServicePrincipalSignInLogs.yaml:18]: @KustoKing Are these changes due to positions changing, is there anything added new or got removed?
Other than EquivalentBuiltInParser property?

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationAADSigninLogs.yaml [hunk 3]
```diff
@@ -15,23 +15,27 @@ References:
 Description: |
   This ASIM parser supports normalizing Azure Active Directory Interactive sign in logs, stored in the  SigninLogs table, to the ASIM Authentication schema.
 ParserName: ASimAuthenticationSigninLogs
+EquivalentBuiltInParser: _ASim_Authentication_SigninLogs
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationAADSigninLogs.yaml:18]: @KustoKing Are these changes due to positions changing, is there anything added new or got removed?
Other than EquivalentBuiltInParser property?

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationM365Defender.yaml [hunk 4]
```diff
@@ -15,6 +15,7 @@ References:
 Description: |
   This ASIM parser supports normalizing endpoint authentication events, collected by Microsoft 365 Defender for Endpoint, stored in the  DeviceLogonEvents table, to the ASIM Authentication schema.
 ParserName: ASimAuthenticationM365Defender
+EquivalentBuiltInParser: _ASim_Authentication_M365Defender
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationM365Defender.yaml:18]: @KustoKing Are these changes due to positions changing, is there anything added new or got removed?
Other than EquivalentBuiltInParser property?

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationMicrosoftMD4IoT.yaml [hunk 5]
```diff
@@ -15,48 +15,49 @@ References:
 Description: |
   This ASIM parser supports normalizing Microsoft Defender for IoT endpoint logs to the ASIM Authentication schema.
 ParserName: ASimAuthenticationMD4IoT
+EquivalentBuiltInParser: _ASim_Authentication_MD4IoT
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationMicrosoftMD4IoT.yaml:18]: @KustoKing Are these changes due to positions changing, is there anything added new or got removed?
Other than EquivalentBuiltInParser property?

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationMicrosoftWindowsEvent.yaml [hunk 6]
```diff
@@ -15,6 +15,7 @@ References:
 Description: |
   This ASIM parser supports normalizing Windows Authentication events (4624, 4625, 4634, and 4647), collected either by the Log Analytics Agent or the Azure Monitor Agent, into either the WindowsEvent (WEF) or SecurityEvent tables, to the ASIM Authentication schema.
 ParserName: ASimAuthenticationMicrosoftWindowsEvent
+EquivalentBuiltInParser: _ASim_Authentication_WindowsEvent
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationMicrosoftWindowsEvent.yaml:18]: @KustoKing Are these changes due to positions changing, is there anything added new or got removed?
Other than EquivalentBuiltInParser property?

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationPostgreSQL.yaml [hunk 7]
```diff
@@ -15,160 +15,173 @@ References:
 Description: |
   This ASIM parser supports normalizing PostgreSQL sign in logs to the ASIM Authentication schema.
 ParserName: ASimAuthenticationPostgreSQL
+EquivalentBuiltInParser: _ASim_Authentication_PostgreSQL
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationPostgreSQL.yaml:18]: @KustoKing Are these changes due to positions changing, is there anything added new or got removed?
Other than EquivalentBuiltInParser property?

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSshd.yaml [hunk 8]
```diff
@@ -19,6 +19,7 @@ Notes: |
   - Certain preauth events log unsuccessful authentication negotiation, which have security importance but are not authentication events per se, and therefore are not used by this parser.
   - The parser may support other sshd servers, but this has not been tested. 
 ParserName: ASimAuthenticationSshd
+EquivalentBuiltInParser: _ASim_Authentication_Sshd
```
Comment (vakohl) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationSshd.yaml:22]: @KustoKing Are these changes due to positions changing, is there anything added new or got removed?
Other than EquivalentBuiltInParser property?

**Review by vakohl — CHANGES_REQUESTED** [review]

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationAADManagedIdentity.yaml [hunk 9]
```diff
@@ -24,90 +25,90 @@ ParserQuery: |
       let AADResultTypesLookup = datatable (ResultType:string, EventResultDetails:string, EventType:string, EventResult:string, EventOriginalResultDetails:string, EventSeverity:string)
       [
           "0"     ,""                         ,"Logon"    ,"Success"  ,"", "Informational",
-          "53003" ,"Logon violates policy"    ,"Logon"    ,"Failure"  ,"53003 - BlockedByConditionalAccess", "Low",
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationAADManagedIdentity.yaml:27]: I sorted the columns

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationAADNonInteractive.yaml (hunk 1 above)
Comment (KustoKing) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationAADNonInteractive.yaml:18]: Sorted the columns

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationMicrosoftMD4IoT.yaml (hunk 5 above)
Comment (KustoKing) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationMicrosoftMD4IoT.yaml:18]: Sorted columns

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationMicrosoftWindowsEvent.yaml (hunk 6 above)
Comment (KustoKing) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationMicrosoftWindowsEvent.yaml:18]: Sorted columns

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationPostgreSQL.yaml (hunk 7 above)
Comment (KustoKing) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationPostgreSQL.yaml:18]: Sorted columns and added EventUid

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationAADServicePrincipalSignInLogs.yaml (hunk 2 above)
Comment (KustoKing) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationAADServicePrincipalSignInLogs.yaml:18]: Sorted columns

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationM365Defender.yaml (hunk 4 above)
Comment (KustoKing) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationM365Defender.yaml:18]: Items sorted

**Review by KustoKing — COMMENTED** [review]
All items sorted and 1 and on 1 item I added EventUid

**Comment by v-atulyadav** [comment]
Hi @KustoKing,
Thank you for accommodating the suggested changes! We will review the Pull Request internally and get back to you by shortly.

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSshd.yaml (hunk 8 above)
Comment (KustoKing) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSshd.yaml:22]: Added  property

**Review by vakohl — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**